databricks/koalas/tests/test_accessor.py

```diff
@@ -0,0 +1,139 @@
+#
+# Copyright (C) 2019 Databricks, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+import contextlib
+import unittest
+import warnings
+
+import numpy as np
+import pandas as pd
+
+from databricks import koalas as ks
+from databricks.koalas.testing.utils import assert_produces_warning
+from databricks.koalas.extensions import (
+    register_dataframe_accessor,
+    register_series_accessor,
+    register_index_accessor,
+)
+
+
+@contextlib.contextmanager
+def ensure_removed(obj, attr):
+    """
+    Ensure attribute attached to 'obj' during testing is removed in the end
+    """
+    try:
+        yield
+
+    finally:
+        try:
+            delattr(obj, attr)
+        except AttributeError:
+            pass
+
+
+class CustomAccessor:
+    def __init__(self, obj):
+        self.obj = obj
+        self.item = "item"
+
+    @property
+    def prop(self):
+        return self.item
+
+    def method(self):
+        return self.item
+
+    def check_length(self, col=None):
+        if type(self.obj) == ks.DataFrame or col is not None:
+            return len(self.obj[col])
+        else:
+            try:
+                return len(self.obj)
+            except Exception as e:
+                raise ValueError(str(e))
+
+
+class AccessorTests(unittest.TestCase):
```

Comment:
Could you extend `ReusedSQLTestCase` instead of `unittest.TestCase`, which will manage the underlying Spark instance?
Also, we'd use `AccessorTest` (without `s` at the end of the class name).
nit: `ExtensionTest` might be better?

Reply:
Fine by me! I refactored `AccessorTests` to `ExtensionTest` inheriting `ReusedSQLTestCase`. For consistency's sake, I also renamed the module to `test_extension` instead of `test_accessor`.